src/ring_theory/ideal/over.lean

```diff
@@ -164,6 +166,38 @@ is_scalar_tower.of_algebra_map_eq $ λ x,
 by rw [quotient.algebra_map_eq, quotient.algebra_map_quotient_map_quotient,
        quotient.mk_algebra_map]
 
+lemma pow_nat_degree_le_of_root_of_eisenstein_criterion_mem {x : S} {f : polynomial S}
+  (hroot : is_root f x) (hmo : f.monic) (hdiv : ∀ n < f.nat_degree, f.coeff n ∈ P) :
```

Comment:
How about adding a definition `polynomial.eisenstein_at`, so that you can write `(hfP : f.eisenstein_at P)`?

Reply:
This is exactly what I am proposing [here](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Eiseinstein.20polynomials). These results are in some sense preliminary, since the polynomial is not really Eiseinstein (there is no need for the condition on the constant term). I can of course state them in the (currently non-existent) file `polynomial.eiseinstein`, but this file will import quite a lot of stuff, since it will also contains nontrivial results, for example about the discriminant.